### PR TITLE
refactor: Add FieldHandler abstraction for template-field transformations

### PR DESCRIPTION
**Pull request by bjchambers**
## Summary

- Introduces a `FieldHandler` ABC with `matches()`, `activate()` (async context manager), and `prepare()` methods to systematize template-field-based parameter transformations
- Migrates three existing inline conversions to modular handlers: `EnvVarFieldHandler`, `StringCoercionFieldHandler`, `DataFrameFieldHandler`
- Adds `_apply_field_handlers` dispatch in `BaseExecutor` that runs handlers sequentially, skipping handlers with no matching fields
- Simplifies `CustomCodeExecutor._prepare_component_parameters` by ~80 lines

## Motivation

The executor's `_prepare_component_parameters` methods contained growing inline logic for dispatching on template field metadata (env var resolution, Message→str coercion, list→DataFrame conversion). This refactoring makes the pattern explicit and extensible — preparing for `FileFieldHandler` for blob-backed file transfer (#617) in a follow-up PR.

## Design

```python
class FieldHandler(ABC):
    def matches(self, template_field: dict) -> bool: ...
    async def activate(self) -> AsyncIterator[Any]:  # context manager, default yields None
        yield None
    async def prepare(self, fields: dict[str, tuple[Any, dict]], context: Any) -> dict[str, Any]: ...
```

- `activate()` is only entered when ≥1 field matches; yields a context value passed to `prepare()` (e.g., temp dir path for future `FileFieldHandler`)
- Handlers run sequentially (ordering may matter); parallel work happens within a handler's `prepare()` via `asyncio.gather`
- Value-based deserialization (`__langflow_type__`, `__class_name__`, `__tool_wrapper__` markers) remains in `_prepare_component_parameters`

## Test plan

- [x] 24 new unit tests for all handlers and `_apply_field_handlers` dispatch
- [x] Updated existing env var tests in `test_base_executor.py` and `test_core_executor.py`
- [x] Full test suite passes (141 passed, 1 skipped)

### CODE REVIEW

File: integrations/langflow/src/stepflow_langflow_integration/executor/field_handlers/dataframe.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+logger = logging.getLogger(__name__)
+
+
+def _is_data_list(value: list[Any]) -> bool:
+    """Check if a list contains Data-like objects suitable for DataFrame conversion."""
+    return all(
+        (isinstance(item, dict) and ("text" in item or "__class_name__" in item))
+        or (hasattr(item, "__class__") and item.__class__.__name__ == "Data")
+        for item in value
+        if item is not None
```
[the remark this answers is not in the document]
Comment (bjchambers) [reply integrations/langflow/src/stepflow_langflow_integration/executor/field_handlers/dataframe.py]: Good catch. Fixed in fcd07b00 with an explicit non_null filter and a test for the all-None edge case.
